app/battle.py

```diff
@@ -0,0 +1,31 @@
+from typing import List
+from app.knights import Knight
+
+
+class BattleKnight:
+    def __init__(self, knights: List[Knight]):
+        self.knights = knights
+
+    def battle_preparation(self) -> None:
+        for knight in self.knights:
+            knight.protection = sum(armour["protection"] for armour in knight.armour)
+            knight.power += knight.weapon["power"]
+
+            if knight.potion:
+                knight.hp += knight.potion["effect"].get("hp", 0)
```

Comment:
The `potion` attribute is assumed to be a dictionary with an `effect` key containing another dictionary. Ensure this structure is correctly defined in the `Knight` class.